### PR TITLE
Vagrantfile to simulate Travis CI container

### PR DESCRIPTION
**Pull request by bfritz**
Configures VirtualBox VM with 2 cores and 4gb memory running
Ubuntu Precise with Oracle JDK8.  Specs are similar to the
Travis CI container environment[1] used for http4s CI builds.

Assumes VirtualBox and Vagrant are already installed.  Then:

    vagrant up    # wait for 1/2 of Internet to download
    vagrant ssh

    # now inside VM
    cd /vagrant
    sbt test

    # ...or to better imitate a particular Travis build
    sbt -v -Djava.security.egd=file:/dev/./urandom 'set scalazVersion in ThisBuild := "7.2.8"' ++2.11.8 ';clean ;test'

[1] https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+# VM configuration below is intended to imitate the Travis CI
+# "container-based" infrastructure where the http4s builds run.
+# https://docs.travis-ci.com/user/ci-environment/#Virtualization-environments
+
+DEFAULT_CPU_COUNT = 2
+DEFAULT_MEMORY = 4096
+EXECUTION_CAP = 100       # set below zero to simulate sluggins Travis build
```
Comment (rossabaker) [inline Vagrantfile]: sluggish

**Comment by bfritz** [comment]
Might be useful for future testing.  On the other hand, it's not very particular to http4s, so maybe a different home is in order.

**Comment by bfritz** [comment]
Using Docker would be more natural given that's how Travis runs it on GCE.  Did you try with docker-compose?
https://docs.docker.com/compose/compose-file/#resources

**Comment by bfritz** [comment]
Going to let this sit as a PR for a bit while I think about where good dev docs would live and maybe experiment with docker-compose.

**Comment by rossabaker** [comment]
I did not.  I got mad and dove back into emacs.

**Comment by rossabaker** [comment]
Now that our CI-is-different-from-local woes seem to be behind us, is this still useful?

**Comment by bfritz** [comment]
Nah.  I don't think so.

**Pull request closed without merging**